src/Type/BitwiseFlagAnalyser.php

```diff
@@ -0,0 +1,99 @@
+<?php declare(strict_types = 1);
+
+namespace PHPStan\Type;
+
+use PhpParser\Node\Expr;
+use PhpParser\Node\Expr\BinaryOp\BitwiseOr;
+use PhpParser\Node\Expr\ConstFetch;
+use PhpParser\Node\Name\FullyQualified;
+use PHPStan\Analyser\Scope;
+use PHPStan\Reflection\ReflectionProvider;
+use PHPStan\TrinaryLogic;
+use PHPStan\Type\Constant\ConstantIntegerType;
+
+final class BitwiseFlagAnalyser
+{
+
+	public function __construct(private ReflectionProvider $reflectionProvider)
+	{
+	}
+
+	public function exprContainsConstant(Expr $expr, Scope $scope, string $constName): TrinaryLogic
+	{
+		if ($expr instanceof ConstFetch) {
+			$resolveConstantName = $this->reflectionProvider->resolveConstantName($expr->name, $scope);
+
+			if ($resolveConstantName !== $constName) {
+				return TrinaryLogic::createNo();
+			}
+		}
+
+		$fqcn = new FullyQualified($constName);
+		if ($this->reflectionProvider->hasConstant($fqcn, $scope)) {
+			$constant = $this->reflectionProvider->getConstant($fqcn, $scope);
+
+			$valueType = $constant->getValueType();
+
+			if ($valueType instanceof ConstantIntegerType) {
+				return $this->exprContainsIntFlag($expr, $scope, $valueType->getValue());
+			}
+		}
+
+		return TrinaryLogic::createNo();
```

Comment:
Even if the constant is not known by the ReflectionProvider, we could support this case for AST-driven analysis. If the expression is ConstFetch of `$constName` or even a bitwise combination where one of the parts is `$constName`, we can return Yes.

Reply:
I am not sure what todo with this feedback.

do you mean this line should be changed to `maybe` and we should add AST-driven analysis with this PR? or in a future PR?

----

when the `ReflectionProvider` doesn't know the constant, and we therefore cannot judge what the value of the constant is (is it a string? is it a int? which constant-int value has it), IMO we cannot analyse whether its contained in a given expression, can we?

Reply:
I think it's just about an `if` that could easily be added. If we ask `BitwiseFlagHelper` about a `FOO_CONSTANT` and we see the AST contains a `ConstFetch` of `FOO_CONSTANT`, we don't need to verify the value because we know it's the same thing.

Reply:
I think I figured it out